libs/sdk-backend-bear/src/convertors/dateFormatting/defaultDateFormatter.ts

```diff
@@ -0,0 +1,21 @@
+// (C) 2020 GoodData Corporation
+import format from "date-fns/format";
+import { UnexpectedError } from "@gooddata/sdk-backend-spi";
+
+import { DateFormatter } from "./types";
+import { dateFormats, DEFAULT_DATE_FORMAT } from "./dateValueParser";
+
+/**
+ * Creates a default date formatting function.
+ * @public
+ */
+export const createDefaultDateFormatter = (dateFormat: string = DEFAULT_DATE_FORMAT): DateFormatter => {
```

Comment:
Suggestion: You could make the `dateFormat` parameter more typesafe by doing this

```ts
// dateValueParser.ts
const dateFormats = ["MM/dd/yyyy", "dd/MM/yyyy", "dd-MM-yyyy", "yyyy-MM-dd", "M/d/yy", "dd.MM.yyyy"] as const;
export type DateFormat = (typeof dateFormats)[number];
export const DEFAULT_DATE_FORMAT: DateFormat = "MM/dd/yyyy";

// defaultDateFormatter.ts
export const createDefaultDateFormatter = (dateFormat: DateFormat = DEFAULT_DATE_FORMAT): DateFormatter => {
//...
}
```

That way, any typos in the format will be detected by TypeScript during compilation (you can try it [here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAJgQygUwGIgE4FskRgXhgG0AiAWTIHo45KBPBukgGhhJsovsZbZoFoK-Rk1YkRgsvxq9y1brJoA6CkpEkAujAR5QkKAG4AUMgAeAB0ywodc8hgARJGkw5YhABQ27IAGbxndGxcAEoiMABXLAAjZAwNYzNLDFg9aEcAUVQAQQBVABkAFQB9B2zCjOLUAHkAJTJygC5HQNckAjYuDnVjEwsrGDTUjGRnB2RfBAiAGygnFCC3FAwOj0QFtqhm+Zdg90ycgpKyiqq6hsKQ5oQwOgIAPhgAbyMYN5gRqAiMMBhI6emxgAvkA)).